lisa/sut_orchestrator/azure/common.py

```diff
@@ -2061,56 +2061,40 @@ def assign_access_policy_to_vm(
     return keyvault_poller.result()
 
 
-def delete_keyvault(
-    platform: "AzurePlatform", vault_name: str, log: Logger, resource_group_name: str
-) -> None:
-    keyvault_client = get_key_vault_management_client(platform)
-
-    keyvault_poller = keyvault_client.vaults.delete(resource_group_name, vault_name)
-    log.debug(f"{keyvault_poller} - Key Vault {vault_name} deleted successfully.")
-
-
-def get_certificate_client(
-    vault_url: str, platform: "AzurePlatform"
-) -> CertificateClient:
-    return CertificateClient(vault_url, platform.credential)
-
-
-def get_secret_client(vault_url: str, platform: "AzurePlatform") -> SecretClient:
-    return SecretClient(vault_url, platform.credential)
-
-
+@retry(tries=5, delay=1)
 def create_certificate(
     platform: "AzurePlatform",
     vault_url: str,
-    log: Logger,
     cert_name: str,
-    retries: int = 5,
-    delay: int = 2,
+    log: Logger,
 ) -> str:
     certificate_client = get_certificate_client(vault_url, platform)
     secret_client = get_secret_client(vault_url, platform)
 
     cert_policy = CertificatePolicy.get_default()
 
-    for _attempt in range(retries):
-        # Create certificate
-        create_certificate_result = certificate_client.begin_create_certificate(
-            cert_name, policy=cert_policy
-        )
-        log.info(
-            f"Certificate '{cert_name}' has been created. "
-            f"Result: {create_certificate_result}"
-        )
-        certificate_client.update_certificate_properties(
-            certificate_name=cert_name, enabled=True
-        )
-        break
+    # Create certificate
+    create_certificate_result = certificate_client.begin_create_certificate(
+        cert_name, policy=cert_policy
+    )
+    log.debug(
+        f"Certificate '{cert_name}' has been created. "
+        f"Result: {create_certificate_result}"
+    )
+    certificate_client.update_certificate_properties(
+        certificate_name=cert_name, enabled=True
+    )
 
     secret_id: Optional[str] = secret_client.get_secret(name=cert_name).id
     if secret_id:
-        secret_url_without_version = secret_id.rsplit("/", 1)[0]
-        return secret_url_without_version
+        match = re.match(r"(https://.+?/secrets/.+?)(?:/[^/]+)?$", secret_id)
```

Comment:
1. provide examples above the regexp patterns, so it's easy to fix, if there is any regression, or new patterns are found.